R/helpers_knitr_Design.R

```diff
@@ -10,17 +10,31 @@
 #' @name knit_print
 NULL
 
-# RuleDesign                     Done
-# DesignGrouped                  Done
-# RuleDesignOrdinal
-# Design                         Done
-# TDsamplesDesign                Done
-# TDDesign                       Done
-# DesignOrdinal                  Done
-# DualDesign                     Done
-# DualResponsesSamplesDesign     Done
-# DualResponsesDesign            Done
-# DADesign                       Done
+#' Internal Helper Functions for Validation of [`StartingDose`] Objects
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' These functions are only used internally to validate the format of an input
+#' [`RuleDesign`] or inherited classes and therefore not exported.
+#'
+#' @name v_starting_dose
+#' @param object (`StartingDose`)\cr object to validate.
+#' @return A `character` vector with the validation failure messages,
+#'   or `TRUE` in case validation passes.
+NULL
```

Comment:
is there a need to have this separate `roxygen` chunk? I wonder if it can be merged into the usual function documentation of `v_starting_dose` below?

Reply:
Will try and see what happens...

Reply:
It looks as if the separation is necessary to prevent a` pkgdown` error.

Reply:
Hm what is the error?

Reply:
I can't remember the exact wording, but it was to do with `v_starting_dose` not being found(?) in the documentation.  That's despite

```
#' @noRd
#' @keywords internal
```

Perhaps `@describeIn` causes a conflict?

Reply:
ok let me have a quick look at this myself

Reply:
I think I could simplify it without error (at least locally `pkgdown` works)